hnn_core/cell.py

```diff
@@ -355,6 +405,57 @@ def __repr__(self):
         class_name = self.__class__.__name__
         return f'<{class_name} | gid={self._gid}>'
 
+    def __eq__(self, other):
+        if not isinstance(other, Cell):
+            return NotImplemented
```

Comment:
needs a test

Reply:
`assert (net == "net") == False.`
Is this test fine?